packages/editor/src/components/start-page-options/index.js

```diff
@@ -27,24 +22,34 @@ export default function StartPageOptions() {
 		);
 		return {
 			postId: getCurrentPostId(),
-			shouldEnable:
+			enabled:
 				choosePatternModalEnabled &&
 				! preferencesModalActive &&
-				! isEditedPostDirty() &&
-				isEditedPostEmpty() &&
 				'page' === getCurrentPostType(),
 		};
 	}, [] );
+	const { isEditedPostDirty, isEditedPostEmpty } = useSelect( editorStore );
 	const { setIsInserterOpened } = useDispatch( editorStore );
 
 	useEffect( () => {
-		if ( shouldEnable ) {
+		if ( ! enabled ) {
+			return;
+		}
+
+		const isFreshPage = ! isEditedPostDirty() && isEditedPostEmpty();
```

Comment:
The effect should react to this state. Check if the page is freshly created when the component is rendered.